Assets/MRTK/Core/Services/MixedRealityToolkit.cs

```diff
@@ -109,6 +116,23 @@ public MixedRealityToolkitConfigurationProfile ActiveProfile
             }
         }
 
+        /// <summary>
+        /// Set the active profile prior to the initialization (i.e. Awake()) of <see cref="MixedRealityToolkit"/>
+        /// </summary>
+        /// <remarks>
+        /// <br>If changing the Active profile during runtime is desired, modify <see cref="ActiveProfile"/> of the active instance directly.</br>
+        /// <br>This function requires the calller script to be executed earlier than <see cref="MixedRealityToolkit"/> script, which can be achieved by setting 
+        /// <see href="https://docs.unity3d.com/Manual/class-MonoManager.html">Script Execution Order settings</see>.</br>
+        /// <br>You are strongly recommended to see 
+        /// <see href="https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/MixedRealityConfigurationGuide.html#changing-profiles-at-runtime">here</see> 
+        /// for more information on profile switching.</br>
+        /// </remarks>
+        public static void SetProfileBeforeInitialization(MixedRealityToolkitConfigurationProfile profile)
+        {
+            MixedRealityToolkit toolkit = FindObjectOfType<MixedRealityToolkit>();
```

Comment:
Should this log an error/warning if this is being called after Awake()?

Reply:
I thought about it as well but as far as I know Unity does not have a built in way of marking whether the Awake of a script has been called or not. I can use a bool to track it but was not sure whether that solution would be elegant enough.

Reply:
Fine either way, just curious what the thought process was